utils/rate_test.go

```diff
@@ -0,0 +1,211 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRateLimit(t *testing.T) {
+	assert := assert.New(t)
+
+	resetAfter := 2 * time.Second
+	mt := &rateTransport{
+		Limit: 3,
+		Reset: time.Now().Add(resetAfter),
+	}
+	c := newClient(assert, mt)
+
+	// first 3 requests must be executed almost immediately
+	spent := measure(func() {
+		c.getSuccess()
+		c.getSuccess()
+		c.getSuccess()
+	})
+	assert.True(spent < time.Millisecond)
+
+	// should sleep before success for more than a second
+	spent = measure(func() {
+		c.getSuccess()
+	})
+	assert.True(spent > 100*time.Millisecond)
```

Comment:
Shouldn't this be nearly `2*time.Second`? Why only `> 100*time.Millisecond`?

Reply:
It's not very different as the real comparison is _immediate_ vs _non-immediate_, but still, maybe it's more informative writing for example `> 1*time.Second` as you did for example [here](https://github.com/src-d/ghsync/pull/61/files#diff-a95f4af213b65e756e16458237fcf615R84).

Reply:
thanks! Fixed. I was playing with smaller numbers and forgot to revert it here. (smaller timeouts make tests unstable sometimes)

Reply:
> smaller timeouts make tests unstable sometimes

Yeah, I came across this once and read something about how the minimum amount of sleep that an OS is able to guarantee.

Reply:
OS sleep interval is much smaller, ~15ms. In this case, the go code was executing too slow when my laptop was under heavy load and it was failing with ~100ms timeouts.